reconcile/terraform_resources.py

```diff
@@ -203,8 +203,10 @@ def get_aws_accounts(
         logging.error(message)
         raise ExcludeAccountsAndDryRunException(message)
 
-    if exclude_accounts and include_accounts:
-        message = "Using --exclude-accounts and --account-name at the same time is not allowed"
+    if (exclude_accounts and include_accounts) and any(
+        a in exclude_accounts for a in include_accounts
```

Comment:
could use `set`

```python
any(set(exclude_accounts) & set(include_accounts))
```

Reply:
i find this less readable, apologies :pray:

Reply:
It's useful when we want to put duplicate result in exception logging like `duplicates := (set(exclude_accounts) & set(include_accounts))`

Not a big issue here as length of both `exclude_accounts` and `include_accounts` should be relative small lists (less than 10), time complexity of `a in exclude_accounts for a in include_accounts` is O(NM) (N = `len(exclude_accounts)`, M = `len(include_accounts`) while using set is O(N+M), in general any N^2 should be avoided in code. Preprocess both to `set` would simplify other checks too, as they all converted to set eventually.